### PR TITLE
Network Scraper Translation to ES Metrics

### PR DESCRIPTION
**Pull request by ishleenk17**
**Description:** This change adds the Network Scraper translations to the ElasticProcessor

**Link to tracking Issue:** https://github.com/elastic/opentelemetry-dev/issues/167

### CODE REVIEW

File: processor/elasticprocessor/internal/hostmetrics/metric.go [hunk 1]
```diff
@@ -48,8 +49,12 @@ func addMetrics(ms pmetric.MetricSlice, resource pcommon.Resource, dataset strin
 		}
 
 		if dataset == "system.process" {
+			// Add resource attribute as an attribute to each datapoint
 			addProcessAttributes(resource, dp)
 		}
+		if metric.attributes != nil {
```
Comment (tommyers-elastic) [inline processor/elasticprocessor/internal/hostmetrics/metric.go:55]: this overwrites the destination map, so needs to come first

File: processor/elasticprocessor/internal/hostmetrics/metric.go [hunk 2]
```diff
@@ -48,8 +49,12 @@ func addMetrics(ms pmetric.MetricSlice, resource pcommon.Resource, dataset strin
...
+			metric.attributes.CopyTo(dp.Attributes())
```
Comment (tommyers-elastic) [inline processor/elasticprocessor/internal/hostmetrics/metric.go]: this overwrites the destination map, so needs to come first

File: processor/elasticprocessor/internal/hostmetrics/network.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+		if metric.Name() == "system.network.io" {
+			dataPoints := metric.Sum().DataPoints()
+			for j := 0; j < dataPoints.Len(); j++ {
+				dp := dataPoints.At(j)
+				if timestamp == 0 {
+					timestamp = dp.Timestamp()
+				}
+				value := dp.IntValue()
+
+				var device string
```
Comment (tommyers-elastic) [inline processor/elasticprocessor/internal/hostmetrics/network.go]: is there any way to factor out line 34-43, 63-72 etc to make it more DRY?

Comment (ishleenk17) [reply processor/elasticprocessor/internal/hostmetrics/network.go]: Done

**Review by tommyers-elastic — APPROVED** [review]

**Pull request merged**